ts/Series/HLC/HLCSeries.ts

```diff
@@ -0,0 +1,497 @@
+/* *
+ *
+ *  (c) 2010-2021 Pawel Lysy
+ *
+ *  License: www.highcharts.com/license
+ *
+ *  !!!!!!! SOURCE GETS TRANSPILED BY TYPESCRIPT. EDIT TS FILE ONLY. !!!!!!!
+ *
+ * */
+
+'use strict';
+
+/* *
+ *
+ *  Imports
+ *
+ * */
+
+import type HLCSeriesOptions from './HLCSeriesOptions';
+import type { StatesOptionsKey } from '../../Core/Series/StatesOptions';
+import type SVGAttributes from '../../Core/Renderer/SVG/SVGAttributes';
+import type SVGElement from '../../Core/Renderer/SVG/SVGElement';
+import type SVGPath from '../../Core/Renderer/SVG/SVGPath';
+import HLCPoint from './HLCPoint.js';
+import SeriesRegistry from '../../Core/Series/SeriesRegistry.js';
+const {
+    seriesTypes: {
+        column: ColumnSeries
+    }
+} = SeriesRegistry;
+import U from '../../Core/Utilities.js';
+const {
+    extend,
+    merge
+} = U;
+
+/* *
+ *
+ *  Class
+ *
+ * */
+
+/**
+ * The hlc series type.
+ *
+ * @private
+ * @class
+ * @name Highcharts.seriesTypes.hlc
+ *
+ * @augments Highcharts.Series
+ */
+class HLCSeries extends ColumnSeries {
```

Comment:
Shouldn't it inherit OHLCSeries?

Reply:
@TorsteinHonsi together with @pawelfus we decided to change the approach. Since the OHLC is series type is an extension of HLC, we decided to change the inharitance order from OHLC -> column to OHLC -> HLC -> column. Some features OHLC would be unaccessable to the HLC (`upColor` for example) so from the architectural point of view this makes more sense. So this way, most of the implementation is moved to HLC, and OHLC series is only responsible for rendering the open value and some stroke handling in terms of options related to `upColor`.